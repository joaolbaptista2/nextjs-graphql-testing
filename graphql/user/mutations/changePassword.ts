import gql from "graphql-tag";

export const changePasswordMutation = gql`
  mutation ChangePassword($data: ChangePasswordInput!) {
    changePassword(data: $data) {
      id,
      name,
      lastName,
      email
    }
}
`;