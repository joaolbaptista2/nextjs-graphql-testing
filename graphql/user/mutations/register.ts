import gql from "graphql-tag";

export const loginMutation = gql`
mutation Register($data: RegisterInput!) {
  register(data: $data) {
    id
    firstName
    lastName
    email
    name
  }
}
`;