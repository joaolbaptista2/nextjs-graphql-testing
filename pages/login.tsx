import React from 'react'
import { Formik, Field } from 'formik'
import { InputField } from '../components/fields/InputField'
import { useLoginMutation, MeQuery } from '../generated/apolloComponents'
import Router from 'next/router'
import { meQuery } from '../graphql/user/queries/me'

const Login = () => {
  const [loginMutation] = useLoginMutation()
  return (
    <Formik onSubmit={async data => {
      console.log(data)
      const response = await loginMutation({
        variables: data,
        update: (cache, {data}) => {
          if (!data) {
            return
          }
          cache.writeQuery<MeQuery>({
            query: meQuery,
            data: {
              __typename: "Query",
              me: data.login
            }
          })
        }
      })
      if (response && response.data && !response.data.login) {
        console.log('Invalid login!')
      }
      Router.push('/')
    }} initialValues={{
      email: '',
      password: ''
    }}>
      {({ handleSubmit  }) => <form data-testid="login-form" onSubmit={handleSubmit}>
        <Field
          data-testid="email-input"
          name="email"
          placeholder="email"
          component={InputField}
        />
        <Field
          data-testid="password-input"
          name="password"
          placeholder="password"
          component={InputField}
          type="password"
        />
        <button type="submit">Login</button>
      </form>}
    </Formik>
  )

}

export default Login