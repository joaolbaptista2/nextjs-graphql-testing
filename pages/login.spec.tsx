import React from 'react'
// import renderer from 'react-test-renderer'
import Login from './login'
import { shallow, mount } from 'enzyme'
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { MockedProvider } from '@apollo/react-testing';
import { loginMutation } from '../graphql/user/mutations/login';


configure({ adapter: new Adapter() });
jest.mock('../pages/login.tsx')
// const tree = renderer.create(
//   <MockedProvider mocks={[]} addTypename={false}>
//     <Login />
//   </MockedProvider>
// )

let wrapper = shallow(
 <MockedProvider mocks={[]} addTypename={false}>
   <Login/>
 </MockedProvider>
)

describe('Login Component', () => {
  it('should not regress', () => {
    expect(wrapper).toMatchSnapshot()
  })

  // Enzyme
  it('should add the correct login data on Submit click', async() => {
    const mocks = [
      {
        request: {
          query: loginMutation,
          variables: {
            email: 'testemail@xxx.com',
            password: 'MyPw12312@',
          },
        },
        result: {
          data: {
            login: {
              id: 1,
              firstName: 'Test',
              lastName: 'Last',
              email: 'testemail@xxx.com',
              name: 'Test Last'
            }
          }
        }
      }
    ]
    const component = await shallow(
      <MockedProvider mocks={mocks} addTypename={false}>
       <Login />
     </MockedProvider>
    )
    // const form = await component.find('Login')
    
    
  })

    //test renderer
  it('should return user on Login', async () => {
  //   const component = await renderer.create(
  //     <MockedProvider mocks={mocks} addTypename={false}>
  //       <Login />
  //     </MockedProvider>
  //   )

  //   const form = component.root.findByType('form');
  //   const emailField = tree.root.findByProps({name: 'email'})
    // emailField.value = 'joao@ustwo.com'
  //   console.log(emailField.props.component(emailField))

    // form.props.onSubmit()

  })
})