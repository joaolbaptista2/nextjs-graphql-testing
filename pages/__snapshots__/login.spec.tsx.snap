// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component should not regress 1`] = `ShallowWrapper {}`;
